src/test/java/com/atlassian/bitbucket/jenkins/internal/fixture/ProjectType.java

```diff
@@ -0,0 +1,55 @@
+package com.atlassian.bitbucket.jenkins.internal.fixture;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Job;
+import hudson.model.TopLevelItem;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+
+import java.util.EnumSet;
+import java.util.Optional;
+
+/**
+ * Describes the kinds of jobs that the {@link JenkinsProjectRule} can create and run tests against. To support another
+ * job type, it should be added to this list, and the project rule should be updated.
+ */
+public enum ProjectType {
```

Comment:
Question: if there's a 1-1 relationship between this enum's values and the Java class type of each project, does this enum really give up anything (other than the slight extra cost of converting back and forth)?

Reply:
The question hinges on the "each type of project" which is unknown. The main reason I wanted an enum as it kills two birds with one stone- it ensure type safety for our supported classes but also is a list of all those supported classes (that we can switch on) so if we ever want to add a new project, we add it as a type to this enum. It's a bit more verbose than just a static list of classes somewhere, but I think it's a bit cleaner.

Reply:
But now I think of it, this could probably be a properties file or something if we prefer

Reply:
I think I'd prefer an enum to a properties file, but I was wondering if the projects' Java types themselves could act as a substitute for the enum, given it'll remove a step of conversion back and forth.
I agree, given the unbounded project Java types (controversial tangent: there isn't a whole lot I miss from Scala, but _sealed traits_ are very useful), I understand solely relying on them can mean that we'll always have to cater for the _unknown/other_ project type, but that also means that by having enums for project types, we're artificially creating a bounded type set, when in reality, it's not guaranteed to stay a closed and immutable set in the future. I think using the project types for all conditional processing would be a bit more honest and transparent, provided we make sure to always cater for that potential "other" type that may be there in the future (e.g. in a future Jenkins version or provided by some new plugin, etc.).

Reply:
Is it a closed immutable set in the future? Can we just add another entry to the enum when we want to expand our set of supported project types?